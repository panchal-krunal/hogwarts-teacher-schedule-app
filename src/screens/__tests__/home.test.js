import React from 'react'
import Enzyme, { shallow ,mount} from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Home from '../home'
Enzyme.configure({ adapter: new Adapter() })

describe('Home component renders correctly', () => {
  it('should render header', () => {
    const wrapper = shallow(<Home />)
    const headerElement  = wrapper.find('[data-test="header-label"]');
    expect(headerElement).toHaveLength(1);
    expect(headerElement.text()).toEqual('Hogwarts Teacher Schedule App');
  })
  it('should render attendance link button', () => {
    const wrapper = shallow(<Home />)
    const buttonElement  = wrapper.find('[data-test="attendance-button"]');
    expect(buttonElement).toHaveLength(1);
    expect(buttonElement.text()).toEqual('Attendance');
  })
  it('should render student schedule button', () => {
    const wrapper = shallow(<Home />)
    const buttonElement  = wrapper.find('[data-test="schedule-button"]');
    expect(buttonElement).toHaveLength(1);
    expect(buttonElement.text()).toEqual('Student Schedule');
  })
})