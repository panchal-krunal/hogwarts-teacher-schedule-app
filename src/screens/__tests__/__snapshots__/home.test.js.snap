// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly should render correctly 1`] = `
<Container
  fluid={false}
>
  <Col>
    <Row>
      <Col
        md="auto"
      >
        <h3
          data-test="header-label"
        >
          Hogwarts Teacher Schedule App
        </h3>
      </Col>
    </Row>
    <Row>
      <Col
        lg="2"
        xs={true}
      >
        <a
          data-test="attendance-button"
          href="/attendance"
        >
          Attendance
        </a>
      </Col>
    </Row>
    <Row>
      <Col
        lg="2"
        xs={true}
      >
        <a
          data-test="schedule-button"
          href="/student-schedule"
        >
          Student Schedule
        </a>
      </Col>
    </Row>
  </Col>
</Container>
`;
